subworkflows/local/generate_downstream_samplesheets/main.nf

```diff
@@ -0,0 +1,112 @@
+                                .tap{ ch_colnames } //ch_header exists using ch_colnames instead
+
+    channelToSamplesheet(ch_colnames, ch_list_for_samplesheet, 'downstream_samplesheets', 'taxprofiler', format, format_sep)
+
+}
+
+workflow SAMPLESHEET_MAG{
+    take:
+    ch_reads
+
+    main:
+    ch_header  = Channel.empty()
+    format     = 'csv' // most common format in nf-core
+    format_sep = ','
+
+    ch_reads
+        .map{meta, reads ->
+                tuple( groupKey(meta.id - "_longReads", 2), meta, reads)
+            }
+        .groupTuple(remainder: true)
+        .map{key, meta, reads ->
+            new_meta = [
+                id: key,
+                run: key,
+                single_end: meta[0].single_end,
+                long_reads: meta[0]?.long_reads ?: meta[1]?.long_reads ?: false
+                ]
+            // Making sure the long reads are the final element of the array.
+            read_files = reads.flatten().sort(false){ a, b -> a.getName().tokenize('.')[0] <=> b.getName().tokenize('.')[0] }
+            [new_meta, read_files]
+            }
+        .tap{ ch_reads_grouped }
+
+    // Make your samplesheet channel construct here depending on your downstream
+    ch_list_for_samplesheet = ch_reads_grouped
+                                .map {
+                                    meta, reads ->
+                                        def out_path       = file(params.outdir).toString() + '/filter/filtered/'
+                                        def sample         = meta.id
+                                        def run            = meta.run
+                                        def group          = ""                                                                                       // only used for co-abundance in binning
+                                        def short_reads_1  = meta.long_reads == (reads.size() > 2) ? out_path + reads[0].getName() : ""                // If long reads, but no short reads, then short_reads_1 is empty
+                                        def short_reads_2  = meta.long_reads == (reads.size() > 2) && reads[1] ? out_path + reads[1].getName() : ""
+                                        def long_reads     = meta.long_reads ? out_path + reads.last().getName() : ""                                  // If long reads, take final element
+                                    [sample: sample, run: run, group: group, short_reads_1: short_reads_1, short_reads_2: short_reads_2, long_reads: long_reads]
+                                }
+                                .tap{ ch_colnames } //ch_header exists using ch_colnames instead
+
+    // Throw a warning that only long reads are not supported yet by MAG
+    ch_list_for_samplesheet
+        .filter{ it.long_reads && it.short_reads_1=="" }
+        .collect{ log.warn("Standalone long reads are not yet supported by the nf-core/mag pipeline and should be REMOVED from the samplesheet \n sample: ${it.sample}" )}
```

Comment:
Why not just remove these (in addition to the warning)?

Reply:
Didn't think people will go into the output.md docs to check for that disclaimer.